### ISSUE TITLE
Header covering up buttons, and unable to click on them even with {force: true}

### ISSUE THREAD

**Issue by jcincotta22** [cypress-io/cypress#537]
* Operating System: Mac OS sierra
* Cypress Version: 19.2
* Browser/Browser Version: Chrome and Electron

### Are you requesting a feature or reporting a bug?
Bug

### Current behavior:
When I try to click on a button at the top of the page it is being covered up by the header.  It essentially disappears from the page once I get the button.  This does not happen outside of the testing framework on any browser.  This is a one page React app.

### Expected behavior:
Able to click the button.

#### Test code:

```js
cy
            .wait("@ageValues")
            // .get("header.MODULES_3gwj0").invoke("hide")
            .wait("@queryHistory3")
            .get("button.MODULES_3b-iD.MODULES_1-U5S.MODULES_snfGC").should("exist").click()
```

### Additional Info (images, notes, stack traces, etc)
```
 1) TriNetX qa-v2 - React Login and Run QuickCount fill in username and password and select a study and run quickCounts:
     CypressError: Timed out retrying: cy.click() failed because this element is being covered by another element:

<a class="MODULES_1uihA" target="_blank" href="http://qa.vpc.trinetx.com/#/signin/?token=AQIC5w.....*&amp;action=admin">Admin</a>

Fix this problem, or use {force: true} to disable error checking.

https://on.cypress.io/element-cannot-be-interacted-with
      at Object.cypressErr (http://qa-v2.vpc.trinetx.com/__cypress/static/js/cypress.js:5723:15)
      at Object.throwErr (http://qa-v2.vpc.trinetx.com/__cypress/static/js/cypress.js:5688:22)
      at Object.throwErrByPath (http://qa-v2.vpc.trinetx.com/__cypress/static/js/cypress.js:5715:21)
      at $Cy._retry (http://qa-v2.vpc.trinetx.com/__cypress/static/js/cypress.js:6663:26)
      at $Cy._waitForAnimations (http://qa-v2.vpc.trinetx.com/__cypress/static/js/cypress.js:8122:25)
      at getCoords (http://qa-v2.vpc.trinetx.com/__cypress/static/js/cypress.js:7431:30)
      at $Cy.retry (http://qa-v2.vpc.trinetx.com/__cypress/static/js/cypress.js:7399:30)
      at http://qa-v2.vpc.trinetx.com/__cypress/static/js/cypress.js:6675:23
      at tryCatcher (http://qa-v2.vpc.trinetx.com/__cypress/static/js/vendor.js:46486:31)
      at Promise._settlePromiseFromHandler (http://qa-v2.vpc.trinetx.com/__cypress/static/js/vendor.js:44621:31)
      at Promise._settlePromiseAt (http://qa-v2.vpc.trinetx.com/__cypress/static/js/vendor.js:44697:18)
      at Promise._settlePromises (http://qa-v2.vpc.trinetx.com/__cypress/static/js/vendor.js:44813:14)
      at Async._drainQueue (http://qa-v2.vpc.trinetx.com/__cypress/static/js/vendor.js:42180:16)
      at Async._drainQueues (http://qa-v2.vpc.trinetx.com/__cypress/static/js/vendor.js:42190:10)
      at MutationObserver.Async.drainQueues (http://qa-v2.vpc.trinetx.com/__cypress/static/js/vendor.js:42018:14)

```

**Comment by jcincotta22** [comment]
[https://files.gitter.im/cypress-io/cypress/CKr6/f122p.mp4](url)

**Comment by jcincotta22** [comment]
![trinetx qa-v2 - react -- login and run quickcount -- fill in username and password and select a study and run quickcounts](https://user-images.githubusercontent.com/19305465/27195176-afc2e0c6-51d3-11e7-94c3-1abe0470a127.png)

**Comment by jcincotta22** [comment]
`cy.get()` method is causing the dom to shift up and elements become blocked.  When using `Cypress.$()` instead of `cy.get()` i am seeing expected behavior.

**Comment by brian-mann** [comment]
`cy.get()` does not scroll or alter the page in any way. It just queries and returns elements.

`.click()` does scroll the page prior to the click. This algorithm is defined in our new docs viewable here:

https://docs-staging.cypress.io/guides/core-concepts/interacting-with-elements.html

**Comment by jcincotta22** [comment]
It should not scroll though...the page is unscrollable.

**Comment by jcincotta22** [comment]
`This most often happens when you have a “sticky nav” that is fixed to the top of the page. By Cypress scrolling the element to the top, it can sometimes end up “underneath” this nav. Our algorithm should detect this and scroll until the element is no longer covered.` this sounds similar to whats happening

**Comment by jcincotta22** [comment]
<img width="1336" alt="screen shot 2017-06-30 at 5 18 57 pm" src="https://user-images.githubusercontent.com/19305465/27754380-4b917f2c-5db8-11e7-9804-17c7bd61b38f.png">

I ran `cy.get(element).debug().click()` and before the click the button is covered up.  You should see the button in the top right of the screen but it is covered up by the nav bar.  Notice there is not a scroll bar anywhere so the page should not be scrolling.

**Comment by brian-mann** [comment]
can you hover over the button in the debugger (so chrome highlights the button)?

Or can you go to inspect element and show us where it is / including the DOM tree?

Likely there is something we need to tweak on our end, but we need a reproducible repo / condition that triggers this behavior.

It's likely just a few conditions that together create the problem. Likely with specific CSS styles, etc. Possibly what would be great is just start yanking out all your CSS / HTML until only the minimum reproduces it

**Comment by jcincotta22** [comment]
<img width="1391" alt="screen shot 2017-06-30 at 5 22 38 pm" src="https://user-images.githubusercontent.com/19305465/27754498-cda48bd0-5db8-11e7-8c17-20b670a54916.png">
this is the page with the buttons not covered up

**Comment by jcincotta22** [comment]
I am unable to hover over the button in debugger

**Comment by brian-mann** [comment]
You can actually force the inspector to highlight the button by doing..

`inspect(subject.get(0))`

That'll highlight the element in the elements panel. Then just hover over that DOM element and chrome should highlight it on your page.

**Comment by jcincotta22** [comment]
<img width="1340" alt="screen shot 2017-06-30 at 5 25 26 pm" src="https://user-images.githubusercontent.com/19305465/27754604-271fc620-5db9-11e7-8498-9217e2ef7b9c.png">
here I grabbed the element with jquery

**Comment by brian-mann** [comment]
Yup okay that's what i was looking for.

**Comment by brian-mann** [comment]
So the page is scrolled prior to the `click`.

And that's the "New Query" button you're trying to click?

**Comment by jcincotta22** [comment]
yes.  There are other buttons too but for this particular part of the test i need that button.

**Comment by brian-mann** [comment]
I see. So something before the `click` is causing the page to be scrolled. And because there's not a scrollbar Cypress is likely failing at scrolling that button into view.

**Comment by jcincotta22** [comment]
And I also ran into another similar problem to this when trying to grab some text on those "activity cards" on the right of the screen.

**Comment by jcincotta22** [comment]
I can probably show you another example if you need one

**Comment by jcincotta22** [comment]
<img width="1320" alt="screen shot 2017-06-30 at 5 34 08 pm" src="https://user-images.githubusercontent.com/19305465/27754823-63202a74-5dba-11e7-9da0-9841ad131976.png">
<img width="1332" alt="screen shot 2017-06-30 at 5 34 19 pm" src="https://user-images.githubusercontent.com/19305465/27754826-69b7f20e-5dba-11e7-9d58-eb67cb833968.png">
 
Notice the dom shifting up after I grab the div.

**Comment by jcincotta22** [comment]
The nav bar is missing because I have to hide it in order to run the test.

**Comment by jcincotta22** [comment]
<img width="1392" alt="screen shot 2017-06-30 at 5 42 52 pm" src="https://user-images.githubusercontent.com/19305465/27755088-a2b29d7e-5dbb-11e7-8227-93c183243874.png">
not sure if this would help too...that header is a fixed header

**Comment by jcincotta22** [comment]
<img width="849" alt="screen shot 2017-06-30 at 5 48 02 pm" src="https://user-images.githubusercontent.com/19305465/27755218-5061ac58-5dbc-11e7-964f-fdd9417964ef.png">

<img width="856" alt="screen shot 2017-06-30 at 5 47 52 pm" src="https://user-images.githubusercontent.com/19305465/27755227-6255d362-5dbc-11e7-88cb-000c9961f5bb.png">

also...in the cypress view it allows me to scroll.

**Comment by kosanna** [comment]
Experiencing the same issue, is there a way just to disable scroll inside cy click/select?

**Comment by brian-mann** [comment]
Okay we're going to disable the scrolling when `{ force: true }` is applied. I see that as the simplest way to work around these issues and I think it just "makes sense".

https://docs.cypress.io/guides/core-concepts/interacting-with-elements.html#Actionability

I'll see if I can get this change out in `0.20.0`.

**Comment by brian-mann** [comment]
I'm pretty sure this is now resolved by fixing #553, and #569. I'll keep this issue open until after `0.20.0` comes out which will enable you to verify it works.

**Comment by brian-mann** [comment]
Actually we've changed our process and are closing issues which already have code that's landed to fix them.

Once we release `0.20.0` please re-open this if you still see this issue.

The code for this is done, but this has yet to be released. We'll update the issue and reference the changelog when it's released.

**Comment by jennifer-shehane** [comment]
This was released in [Cypress 0.20.0](https://docs.cypress.io/guides/references/changelog.html#0-20-0)
